xds/internal/resolver/xds_resolver.go

```diff
@@ -162,7 +162,7 @@ const jsonFormatSC = `{
     "loadBalancingConfig":[
       {
         "experimental_cds":{
-          "Cluster": %s
+          "Cluster": "%s"
```

Comment:
Does `json.Unmarshal` fail without the quotes? Let's add some sanity check.

If you are adding those in the following PR, it's OK to submit this as is.

Reply:
Yes `json.Unmarshal` fails without the quotes. In the other test PR, I verify that the service config is parsed without an error by the cdsBuilder. It is an indirect test though. There doesn't seem to be an easy way to directly test if `jsonFormatSC` contains a valid JSON.

Specifically, handling this TODO was how I uncovered this issue:
https://github.com/grpc/grpc-go/blob/master/xds/internal/resolver/xds_resolver_test.go#L74

Reply:
The test cleanup PR is here:
https://github.com/grpc/grpc-go/pull/3233

And it fails as follows without the fix in here::

    easwars@golang:~/src/grpc-go$ go test -count=1 -race -cpu 4 -timeout=50s 
    google.golang.org/grpc/xds/internal/resolver/...
    --- FAIL: TestXDSResolverGoodServiceUpdate (0.00s)
        xds_resolver_test.go:317: ClientConn.UpdateState received error in service config: invalid character 'c' looking for beginning of value
    FAIL
    FAIL    google.golang.org/grpc/xds/internal/resolver    1.203s